pallets/connectors-gateway/src/lib.rs

```diff
@@ -46,26 +40,47 @@ pub mod pallet {
 	#[pallet::generate_store(pub (super) trait Store)]
 	pub struct Pallet<T>(_);
 
+	#[pallet::origin]
+	pub type Origin = GatewayOrigin;
+
 	#[pallet::config]
-	pub trait Config: frame_system::Config + pallet_xcm_transactor::Config {
-		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+	pub trait Config: frame_system::Config {
+		/// The origin type.
+		type RuntimeOrigin: Into<Result<GatewayOrigin, <Self as frame_system::Config>::RuntimeOrigin>>
+			+ From<GatewayOrigin>;
 
-		type WeightInfo: WeightInfo;
+		/// The event type.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
 
 		/// The LocalOrigin ensures that some calls can only be performed from a
 		/// local context i.e. a different pallet.
-		type LocalOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = DomainAddress>;
+		type LocalOrigin: EnsureOrigin<
+			<Self as frame_system::Config>::RuntimeOrigin,
+			Success = DomainAddress,
+		>;
 
 		/// The AdminOrigin ensures that some calls can only be performed by
 		/// admins.
-		type AdminOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+		type AdminOrigin: EnsureOrigin<<Self as frame_system::Config>::RuntimeOrigin>;
 
 		/// The incoming and outgoing message type.
 		type Message: Codec;
 
+		/// The message router type that is stored for each domain.
+		type Router: DomainRouter<Sender = Self::AccountId, Message = Self::Message>
+			+ Clone
+			+ Debug
+			+ MaxEncodedLen
+			+ TypeInfo
+			+ FullCodec
+			+ EncodeLike
+			+ PartialEq;
+
 		/// The type that processes incoming messages.
 		type Connectors: InboundQueue<Sender = Domain, Message = Self::Message>;
 
+		type WeightInfo: WeightInfo;
+
 		/// Maximum size of an Ethereum message.
 		#[pallet::constant]
 		type MaxEthMsg: Get<u32>;
```

Comment:
It seems like we don't use this anywhere at the moment. I'd rather not have code around that we don't use, or we it is where we should. Also, can we [add the units to the naming](https://ruudvanasseldonk.com/2022/03/20/please-put-units-in-names) please :) ?

Reply:
Renamed to `MaxEthMsgSize`, and it is used in `process_msg`.

Reply:
> and it is used in `process_msg`.

Thanks, I missed that!

So, basically this is used to represent the max size of a Connectors message; More specifically, incoming Connectors messages, although I don't think the specialisation is important.

With that in mind, to avoid confusing the `Ethereum` transaction message with the Connectors message sent through an Ethereum transaction, I suggest that we name this `MaxMsgBytes` with an appropriate `///` comment.

Alternative, I wonder if we can just use the `MaxEncodedLength` type of the `Message` type? That would make the code more resilient (since a change to the Message enum type wouldn't require changing this type here) reduce the sources of truth about the longest message byte length.

Reply:
Hmm, I think there's some confusion here. This should be the maximum size of the eth message that eventually gets deserialized into a connectors message.

Reply:
I don't think so? Following to the code, it is the `Connectors message`:

``` rust
               pub fn process_msg(
			origin: OriginFor<T>,
			msg: BoundedVec<u8, T::MaxEthMsgSize>, // HERE
		) -> DispatchResult {
			let domain_address = T::LocalOrigin::ensure_origin(origin)?;

			match domain_address {
				DomainAddress::EVM(_, _) => {
					DomainSubmitters::<T>::get(domain_address.domain())
						.iter()
						.find(|s| s.eq(&&domain_address))
						.ok_or(Error::<T>::UnknownSubmitter)?;

                                         // AND HERE
					let incoming_msg = T::Message::deserialize(&mut msg.as_slice())
						.map_err(|_| Error::<T>::EthereumMessageDecode)?; 
```

Reply:
My bad, I'm still confused as to how this would be called then. Wouldn't it make more sense for the caller to just pass a connectors message so that we can avoid deserializing here?

Reply:
Yap :)